src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/TypeMetadataMapNode.cs

```diff
@@ -76,7 +76,12 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
 
             foreach (var mappingEntry in factory.MetadataManager.GetTypeDefinitionMapping())
             {
-                var node = (EETypeNode)factory.ConstructedTypeSymbol(mappingEntry.Entity);
+                var node = factory.ConstructedTypeSymbol(mappingEntry.Entity) as EETypeNode;
+
+                // ConstructedTypeSymbols can be ExternSymbolNodes if they're defined in another module
```

Comment:
This is an odd check.  We should just consult the compilation group whether the type is in the compilation group before asking for the ConstructedTypeSymbol.